requests/models.py

```diff
@@ -404,7 +404,8 @@ def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
         if headers:
-            self.headers = CaseInsensitiveDict(validate_header(i) for i in headers.items())
+            valid_headers = filter(is_valid_header, headers.items())
+            self.headers = CaseInsensitiveDict((to_native_string(name), val) for name, val in valid_headers)
```

Comment:
I'm not particularly fond of having to iterate over the header list twice, but I couldn't devise a way to wrap name in to_native_string() while using `is_valid_header` as a filter function. Perhaps I'm missing an easy solution though.